tests/e2e/device-plugin_test.go

```diff
@@ -27,168 +27,154 @@ import (
 
 	helm "github.com/mittwald/go-helm-client"
 	helmValues "github.com/mittwald/go-helm-client/values"
-	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
-	extclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/rand"
 
-	"github.com/NVIDIA/k8s-device-plugin/tests/e2e/common"
-	"github.com/NVIDIA/k8s-device-plugin/tests/e2e/common/diagnostics"
-	"github.com/NVIDIA/k8s-device-plugin/tests/e2e/framework"
+	"github.com/NVIDIA/k8s-device-plugin/tests/e2e/internal"
+	"github.com/NVIDIA/k8s-test-infra/pkg/diagnostics"
 )
 
 const (
 	devicePluginEventuallyTimeout = 10 * time.Minute
 )
 
 // Actual test suite
-var _ = NVDescribe("GPU Device Plugin", func() {
-	f := framework.NewFramework("k8s-device-plugin")
-
-	Context("When deploying k8s-device-plugin", Ordered, func() {
-		// helm-chart is required
-		if *HelmChart == "" {
-			Fail("No helm-chart for k8s-device-plugin specified")
+var _ = Describe("GPU Device Plugin", Ordered, Label("gpu", "e2e", "device-plugin"), func() {
+	// Init global suite vars vars
+	var (
+		helmReleaseName string
+		chartSpec       helm.ChartSpec
+
+		collectLogsFrom      []string
+		diagnosticsCollector *diagnostics.Diagnostic
+	)
+
+	defaultCollectorObjects := []string{
+		"pods",
+		"nodes",
+		"namespaces",
+		"deployments",
+		"daemonsets",
+		"jobs",
+	}
+
+	values := helmValues.Options{
+		Values: []string{
+			fmt.Sprintf("image.repository=%s", ImageRepo),
+			fmt.Sprintf("image.tag=%s", ImageTag),
+			fmt.Sprintf("image.pullPolicy=%s", ImagePullPolicy),
+			"devicePlugin.enabled=true",
+			// We need to make affinity is none if not deploying NFD/GFD
+			// test will fail if not run on a GPU node
```

Comment:
Are we sure that this test runs separately from GFD / NFD? Is an alternative not to EXPLICITLY label the note with `nvidia.com/gpu.present=true`?

Reply:
yes, the test runs in a serial mode, running device plugin first and then GFD tests. By setting `affinity` to null we don't need node labels.